packages/xo-server/src/vhd-merge.integ.spec.js

```diff
@@ -0,0 +1,203 @@
+/* eslint-env jest */
+
+import execa from 'execa'
+import fs from 'fs-extra'
+import path from 'path'
+import rimraf from 'rimraf'
+import LocalHandler from './remote-handlers/local.js'
+import { tmpDir, pFromCallback } from './utils'
+import vhdMerge, { chainVhd, Vhd } from './vhd-merge'
+
+const initialDir = process.cwd()
+
+jest.setTimeout(10000)
+
+beforeEach(async () => {
+  const dir = await tmpDir()
+  process.chdir(dir)
+})
+
+afterEach(async () => {
+  const tmpDir = process.cwd()
+  process.chdir(initialDir)
+  await pFromCallback(cb => rimraf(tmpDir, cb))
+})
+
+async function createRandomFile (name, sizeMo) {
+  await execa('bash', [
+    '-c',
+    `< /dev/urandom tr -dc "\\t\\n [:alnum:]" | head -c ${sizeMo}M >${name}`,
+  ])
+}
+
+async function checkFile (vhdName) {
+  await execa('vhd-util', ['check', '-p', '-b', '-t', '-n', vhdName])
+}
+
+async function recoverRawContent (vhdName, rawName, originalSize) {
+  await checkFile(vhdName)
+  await execa('qemu-img', ['convert', '-fvpc', '-Oraw', vhdName, rawName])
+  await execa('truncate', ['-s', originalSize, rawName])
+}
+
+async function convertFromRawToVhd (rawName, vhdName) {
+  await execa('qemu-img', ['convert', '-f', 'raw', '-Ovpc', rawName, vhdName])
+}
+
+test('writeData on empty file', async () => {
+  expect.assertions(1)
+  const moOfRandom = 11
+  await createRandomFile('randomfile', moOfRandom)
+  await execa('qemu-img', ['create', '-fvpc', 'empty.vhd', moOfRandom + 'M'])
+  const randomData = await fs.readFile('randomfile')
+  const handler = new LocalHandler({ url: 'file://' + process.cwd() })
+  const originalSize = await handler.getSize('randomfile')
+  const newVhd = new Vhd(handler, 'empty.vhd')
+  await newVhd.readHeaderAndFooter()
+  await newVhd.readBlockTable()
+  await newVhd.writeData(0, randomData)
+  await recoverRawContent('empty.vhd', 'recovered', originalSize)
+  expect(await fs.readFile('recovered')).toEqual(randomData)
+})
+
+test('writeData in 2 non-overlaping operations', async () => {
+  expect.assertions(1)
+  const moOfRandom = 11
+  await createRandomFile('randomfile', moOfRandom)
+  await execa('qemu-img', ['create', '-fvpc', 'empty.vhd', moOfRandom + 'M'])
+  const randomData = await fs.readFile('randomfile')
+  const handler = new LocalHandler({ url: 'file://' + process.cwd() })
+  const originalSize = await handler.getSize('randomfile')
+  const newVhd = new Vhd(handler, 'empty.vhd')
+  await newVhd.readHeaderAndFooter()
+  await newVhd.readBlockTable()
+  const splitPointSectors = 4
+  await newVhd.writeData(0, randomData.slice(0, splitPointSectors * 512))
```

Comment:
Why not using the `VHD_SECTOR_SIZE` constant for clearer code?

Reply:
It's not exposed here. We need to expose some vhd things cleanly in a, interface file, but knowing that we were two people working on the same subject, I didn't want to suffer a nightmarish wide range merge.

Reply:
I see, do you mind doing this now?

Reply:
I would like to grab your VHD stream first (it's not in this PR), then I think we have to decide what to do with vhd-cli and xo-vmdk-to-vhd (probably put them under test), lastly decide the new package dependency graph.